src/main/java/teammates/ui/controller/Action.java

```diff
@@ -238,10 +238,9 @@ protected AccountAttributes authenticateAndGetNominalUser(UserType loggedInUserT
                 
                 throw new UnauthorizedAccessException("Unregistered user for a page that needs one");
             } else if (isPageNotCourseJoinRelated() && doesRegkeyBelongToUnregisteredStudent() && isUserLoggedIn) {
-                Url redirectUrl = new Url(Const.ViewURIs.LOGOUT)
-                                      .withParam(Const.ParamsNames.NEXT_URL, requestUrl)
-                                      .withUserId(StringHelper.encrypt(account.googleId)); 
-                
+                Url redirectUrl = new Url(student.getRegistrationUrl())
+                                      .withParam(Const.ParamsNames.STUDENT_NEW_COURSE, student.course)
+                                      .withParam(Const.ParamsNames.NEXT_URL, requestUrl);
```

Comment:
Ok, so now here is how it works: We know that if this code block is reached, these conditions must be fulfilled:
1. There is a logged in user
2. The user is unregistered for the new course
3. The user is registered in some other course(s)
4. The user navigates to a URL other than the join URL
In this case we pass the ID of the new course as a parameter. The existence of this parameter will then decide whether the page looks like this (if exists):
![image](https://cloud.githubusercontent.com/assets/7261051/11361146/1def37da-92c6-11e5-8b46-7217ffe1bfd8.png)
or this (if doesn't exist):
![image](https://cloud.githubusercontent.com/assets/7261051/11204567/36907456-8d3b-11e5-9d31-d352cdfd06ee.png)
This should cover all cases. Note that the difference is purely cosmetic. On both pages' counterparts, the buttons will direct to the same URL.